dotCMS/src/integration-test/java/com/dotcms/rest/api/v3/contenttype/FieldResourceTest.java

```diff
@@ -458,6 +459,137 @@ public void shouldMoveFields () throws DotSecurityException, DotDataException {
         checkAllFieldsIds(listExpected, contentTypeFromDB.fields());
     }
 
+
+    /**
+     * When try to move fields in a Content Type with a right layout
+     * Should update the field with the new sortOrder
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void testSerializerMoveFields () throws DotSecurityException, DotDataException {
+
+        final ContentType type = createContentType();
+        final List<Map<String, Object>> fieldList = createTestFields (type.id());
+        final MoveFieldsForm form     = new MoveFieldsForm(fieldList);
+
+        final FieldLayout fieldLayout = form.getRows(type);
+
+        assertNotNull(fieldLayout);
+        assertNotNull(fieldLayout.getFields());
+        assertEquals(5, fieldLayout.getFields().size());
+        assertTrue(fieldLayout.getFields().get(0) instanceof RowField);
+        assertTrue(RowField.class.cast(fieldLayout.getFields().get(0)).dataType() == DataTypes.SYSTEM);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use assertSame(x, y) instead of assertTrue(x==y), or assertNotSame(x,y) vs assertFalse(x==y)](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5214528)